### PR TITLE
Documentation pattern to Component

### PR DESCRIPTION
**Pull request by freddykrunn**

### CODE REVIEW

File: generators/component/templates/component.ts [hunk 1]
```diff
@@ -46,6 +88,11 @@ export class <%= component.class %> extends<% if (component.isExtendingMes) { %>
 
     //#endregion
 
+    /**
+     * Constructor
+     * 
+     * @param viewContainerRef the reference to the component view container
+     */
     constructor(viewContainerRef: ng.ViewContainerRef) {
```
Comment (jpsfs) [inline generators/component/templates/component.ts:96]: Everytime I need to use a progress indicator, if I try to use the viewContainerRef.elementRef it doesn't work. So my vote is to use get by default the elemenRef and not the viewContainerRef.

**Review by jpsfs — APPROVED** [review]

File: generators/component/templates/component.ts (hunk 1 above)
Comment (pesimeao) [reply generators/component/templates/component.ts:96]: @jpsfs also the elementRef is stores in CoreComponent.framework.elementRef, to be used by customization if needed. This should be changed to elementRef anyway.

Comment (soulcorrosion) [reply generators/component/templates/component.ts:96]: @jpsfs @pesimeao Different concerns gentlemen. If it does not give you the elementRef to use in the progress indicator, either inject it yourself or add it here as well.
ViewContainerRef is a preparation for CMFFLex use as it has more leverage than ElementRef.

File: generators/component/templates/component.ts [hunk 2]
```diff
@@ -10,21 +11,62 @@ import * as Mes from "cmf.mes/src/mes";
...
- * Please provide a meaningful description of this component and how to use it
+ * ## Component name
+ * 
+ * Please provide a meaningful description of this component
+ * Try to answer these questions: 
+ * What is it?
+ * What does it do?
+ * Where to use it?
+ * How to use it?
+ * How it behaves with different sizes?
```
Comment (francisco-maciel) [inline generators/component/templates/component.ts:33]: The grammar is not quite right on this one, can you change this line to _"How does it behave with different sizes?"_

File: generators/widget/templates/widget.ts [hunk 3]
```diff
@@ -27,25 +27,26 @@ let outputs: Map<string, Widget.WidgetOutput> = new Map<string, Widget.WidgetOut
  * * What does it do?
  * * Where to use it?
  * * How to use it?
- * 
+ * * How it behave with different sizes?
```
Comment (francisco-maciel) [inline generators/widget/templates/widget.ts:30]: Same as above, change to _"How does it behave with different sizes?"_

**Pull request merged**